homeassistant/components/home_connect/config_flow.py

```diff
@@ -45,9 +46,16 @@ async def async_step_reauth_confirm(
 
     async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
         """Create an oauth config entry or update existing entry for reauth."""
+        await self.async_set_unique_id(
+            jwt.decode(
+                data["token"]["access_token"], options={"verify_signature": False}
+            )["sub"]
+        )
         if self.source == SOURCE_REAUTH:
-            return self.async_update_reload_and_abort(
-                self._get_reauth_entry(),
-                data_updates=data,
-            )
+            reauth_entry = self._get_reauth_entry()
+            if self.unique_id == reauth_entry.unique_id:
+                return self.async_update_reload_and_abort(
+                    reauth_entry, data_updates=data
+                )
```

Comment:
Use `self._abort_if_unique_id_mismatch(reason="wrong_account")` instead, I use this in withings

Reply:
Will that create a new account?

Reply:
Never mind, I understood it incorrectly